services/QuillLMS/app/services/student_learning_sequences/handle_assignment.rb

```diff
@@ -2,65 +2,52 @@
+      @user_id = user_id
+      @backfill = backfill
     end
 
-    def run
-      return unless student_learning_sequence
-
-      assign_activities_to_sequence
-    end
+    def run = assign_activities_to_sequence
 
     private def assign_activities_to_sequence
       activities.map do |activity|
-        StudentLearningSequenceActivity.find_or_create_by(activity:,
-          classroom_unit:,
-          student_learning_sequence:)
-      end
-    end
+        student_learning_sequence = FindStudentLearningSequence.run(activity.id, classroom_unit_id, user_id)
 
-    private def activities = classroom_unit.unit_activities.map(&:activity)
-    private def classroom_unit = @classroom_unit ||= ClassroomUnit.find(classroom_unit_id)
-    private def initial_activity = @initial_activity ||= FindInitialActivity.run(activities&.first&.id, classroom_unit_id)
-    private def initial_classroom_units = @initial_classroom_units ||= fetch_initial_classroom_units
-    private def pre_diagnostic = @pre_diagnostic ||= activities.find { |a| !a.follow_up_activity_id.nil? }
-    private def student = @student ||= User.find(student_id)
-    private def user_id = student_id
+        # If there isn't a base StudentLearningSequence, we assume this activity is not part of a sequence
+        next unless student_learning_sequence
 
-    private def student_learning_sequence
-      @student_learning_sequence ||= fetch_student_learning_sequence ||
-                                     create_student_learning_sequence
-    end
+        student_learning_sequence_activity = StudentLearningSequenceActivity.create_or_find_by({
+          activity:,
+          classroom_unit:,
+          student_learning_sequence:
+        })
 
-    private def fetch_initial_classroom_units
-      return ClassroomUnit.where(id: classroom_unit_id) if pre_diagnostic
+        if backfill
+          student_learning_sequence.update(**backfill_params)
+          student_learning_sequence_activity.update(**backfill_params)
+        end
 
-      ClassroomUnit.joins(unit: { unit_template: :activities })
-        .where(unit: { unit_template: { activities: initial_activity } })
+        student_learning_sequence_activity
+      end
     end
 
-    private def fetch_student_learning_sequence
-      StudentLearningSequence.order(created_at: :desc)
-        .find_by(
-          initial_classroom_unit: initial_classroom_units,
-          initial_activity:,
-          user_id:
-        )
+    private def classroom_unit = @classroom_unit ||= ClassroomUnit.find(classroom_unit_id)
+
+    private def activities
+      Activity.joins(unit_activities: { unit_unscoped: :classroom_units })
+        .where(classroom_units: { id: classroom_unit_id })
     end
 
-    private def create_student_learning_sequence
-      # Do not create new sequences unless we're processing a pre diagnostic assignment
-      return unless pre_diagnostic
+    private def backfill_params
+      return {} unless backfill
```

Comment:
This feels redundant given the `if backfill` earlier.

Reply:
Good call-out.  That's a leftover from an earlier version of the code.